tests/handlers/test_directory.py

```diff
@@ -432,6 +432,60 @@ def test_allowed(self):
         self.assertEquals(200, channel.code, channel.result)
 
 
+class TestCreatePublishedRoomACL(unittest.HomeserverTestCase):
+    user_id = "@admin:test"
```

Comment:
There's code which [treats admin users differently](https://github.com/matrix-org/synapse/blob/b10257e87972d158f4b6a0c7d1fe7239014ea10a/synapse/handlers/room.py#L644), but this isn't properly registering the user as an admin. I think change this to a different name, say `@test2:test`.

Also suggest calling this `accepted_user_id` instead of just `user_id`.